swift/insecure-url-host-hassuffix-check.yaml

```diff
@@ -0,0 +1,65 @@
+rules:
+  - id: insecure-url-host-hassuffix-check
+    message: >
+      Potentially insecure URL host suffix check. Using hasSuffix("domain.com") without 
+      a leading dot can match unintended domains like "fakedomain.com"
+    languages: [swift]
+    severity: WARNING
+    metadata:
+      category: security
+      cwe: CWE-697
+      technology: [swift, ios]
+      subcategory: [audit]
+      confidence: HIGH
+      likelihood: HIGH
+      impact: MEDIUM
+      references:
+        - https://developer.apple.com/documentation/foundation/url/host(percentencoded:)
+        - https://developer.apple.com/documentation/foundation/nsstring/hassuffix(_:)
+        - https://pentesterlab.com/blog/rust-cors-vulnerabilities
+    pattern-either:
+      # Pattern 1: Any .host property followed by hasSuffix
+      - patterns:
+          - pattern: $X.hasSuffix($SUFFIX)
```

Comment:
If you use a [string literal metavariable](https://semgrep.dev/docs/writing-rules/pattern-syntax#literal-metavariables) (e.g. `"$SUFFIX"`) here that will simplify your regex below a little bit.

Looks like Swift only uses double quotes for string literals, so it's a bit less effective, but this is extra nice in languages that allow both types of quotes for string literals. Then the string literal metavariables will catch both types without needing fancy regexes.